lib/launcher.ts

```diff
@@ -108,197 +107,170 @@ let initFn = function(configFile: string, additionalConfig: Config) {
-                      config.multiCapabilities = multiCapabilities;
-                      config.capabilities = null;
-                    })
-                    .then(() => {
-                      resolve();
-                    })
-                    .catch(err => {
-                      reject(err);
-                    });
-              } else {
-                resolve();
-              }
-            })
-            .then(() => {
-              // 2) Set `multicapabilities` using `capabilities`,
-              // `multicapabilities`,
-              // or default
-              if (config.capabilities) {
-                if (config.multiCapabilities.length) {
-                  logger.warn(
-                      'You have specified both capabilities and ' +
-                      'multiCapabilities. This will result in capabilities being ' +
-                      'ignored');
-                } else {
-                  // Use capabilities if multiCapabilities is empty.
-                  config.multiCapabilities = [config.capabilities];
-                }
-              } else if (!config.multiCapabilities.length) {
-                // Default to chrome if no capabilities given
-                config.multiCapabilities = [{browserName: 'chrome'}];
-              }
-            });
-      })
-      .then(() => {
-        // 3) If we're in `elementExplorer` mode, run only that.
-        if (config.elementExplorer || config.framework === 'explorer') {
-          if (config.multiCapabilities.length != 1) {
-            throw new Error('Must specify only 1 browser while using elementExplorer');
-          } else {
-            config.capabilities = config.multiCapabilities[0];
-          }
-          config.framework = 'explorer';
+  // 2) Set `multicapabilities` using `capabilities`,
+  // `multicapabilities`, or default
+  if (config.capabilities) {
+    if (config.multiCapabilities.length) {
+      logger.warn(
+          'You have specified both capabilities and ' +
+          'multiCapabilities. This will result in capabilities being ' +
+          'ignored');
+    } else {
+      // Use capabilities if multiCapabilities is empty.
+      config.multiCapabilities = [config.capabilities];
+    }
+  } else if (!config.multiCapabilities.length) {
+    // Default to chrome if no capabilities given
+    config.multiCapabilities = [{browserName: 'chrome'}];
+  }
 
-          let runner = new Runner(config);
-          return runner.run().then(
-              (exitCode: number) => {
-                process.exit(exitCode);
-              },
-              (err: Error) => {
-                logger.error(err);
-                process.exit(1);
-              });
-        }
-      })
-      .then(() => {
-        // 4) Run tests.
-        let scheduler = new TaskScheduler(config);
+  // 3) If we're in `elementExplorer` mode, run only that.
+  if (config.elementExplorer || config.framework === 'explorer') {
```

Comment:
ElementExplorer won't work without the control flow. If it's enabled, we should just log an error and exit.

Reply:
Done.